sdk/search/Azure.Search.Documents/src/AzureSearchDocumentsEventSource.cs

```diff
@@ -15,35 +18,69 @@ internal sealed class AzureSearchDocumentsEventSource : EventSource
 
         private AzureSearchDocumentsEventSource() :
             base(EventSourceName, EventSourceSettings.Default, AzureEventSourceListener.TraitName, AzureEventSourceListener.TraitValue)
         { }
 
         public static AzureSearchDocumentsEventSource Instance { get; } = new AzureSearchDocumentsEventSource();
 
-        [Event(PendingQueueResizedEvent, Level = EventLevel.Verbose, Message = "Indexing publisher pending queue size = {0}.")]
-        public void PendingQueueResized(int queueSize)
+        [Event(PendingQueueResizedEvent, Level = EventLevel.Verbose, Message = "{0}: pending queue size = {1}.")]
+        public void PendingQueueResized(string publisherType, int queueSize)
         {
-            WriteEvent(PendingQueueResizedEvent, queueSize);
+            WriteEvent(PendingQueueResizedEvent, publisherType, queueSize);
         }
 
-        [Event(RetryQueueResizedEvent, Level = EventLevel.Verbose, Message = "Indexing publisher retry queue size = {0}.")]
-        public void RetryQueueResized(int queueSize)
+        [Event(RetryQueueResizedEvent, Level = EventLevel.Verbose, Message = "{0}: retry queue size = {1}.")]
+        public void RetryQueueResized(string publisherType, int queueSize)
         {
-            WriteEvent(RetryQueueResizedEvent, queueSize);
+            WriteEvent(RetryQueueResizedEvent, publisherType, queueSize);
         }
 
-        [Event(BatchSubmittedEvent, Level = EventLevel.Informational, Message = "Indexing publisher at {0} has submitted batch of size {1}.")]
-        public void BatchSubmitted(string endPoint, int batchSize)
+        [Event(BatchSubmittedEvent, Level = EventLevel.Informational, Message = "{0} at {1} has submitted batch of size {2}.")]
+        public void BatchSubmitted(string publisherType, string endPoint, int batchSize)
         {
-            WriteEvent(BatchSubmittedEvent, endPoint, batchSize);
+            WriteEvent(BatchSubmittedEvent, publisherType, endPoint, batchSize);
         }
 
-        [Event(BatchActionCountUpdatedEvent, Level = EventLevel.Informational, Message = "Indexing publisher at {0} has updated the starting batch action count from {1} to {2}.")]
-        public void BatchActionCountUpdated(string endPoint, int oldBatchCount, int newBatchCount)
+        [Event(BatchActionCountUpdatedEvent, Level = EventLevel.Warning, Message = "{0} at {1} has updated the starting batch action count from {2} to {3}.")]
+        public void BatchActionCountUpdated(string publisherType, string endPoint, int oldBatchCount, int newBatchCount)
         {
-            WriteEvent(BatchActionCountUpdatedEvent, endPoint, oldBatchCount, newBatchCount);
+            WriteEvent(BatchActionCountUpdatedEvent, publisherType, endPoint, oldBatchCount, newBatchCount);
+        }
+
+        [Event(SearchIndexingBufferedSenderDisposedWithPendingActionsEvent, Level = EventLevel.Error, Message = "{0}: {2} unsent indexing actions at {1}.")]
```

Comment:
Should we change all the `at`s to `for`s?

Reply:
Done.